core/shared/src/test/scala/cats/effect/cps/AsyncAwaitSpec.scala

```diff
@@ -87,19 +87,21 @@ class AsyncAwaitSpec extends Specification with CatsEffect {
     "be cancellable" in {
 
       val program = for {
-        ref <- Ref[IO].of(0)
-        _ <- async[IO] {
+        defer <- Deferred[IO, Unit]
+        fiber <- async[IO] {
+          defer.complete(()).await
```

Comment:
The point of having the never first was to ensure that the following effect (the incrementation) was never actually run after

Reply:
Good point - I can re-add that part to the test together with new assertion.

Reply:
Done